### PR TITLE
TEP-0079: Tekton Catalog Support Tiers [Problem Statement] 

### PR DESCRIPTION
**Pull request by jerop**
This change adds the problem statement for Support Tiers in the Tekton
Catalog - we will explore the proposal in further changes.

The aim of this TEP is to establish support tiers for resources in the
Tekton Catalog to ensure users get high quality resources that they can
rely on while making it easy for contributors to submit resources to the
Tekton Catalog.

The goals of the TEP are to address the following aspects of the Tekton
Catalog:
- Ownership and Maintenance
- Automated Testing and Dogfooding
- Image Scanning for Common Vulnerabilities and Exposures (CVEs)
- Verified Remote Resources

/cc co-authors @vinamra28 @vdemeester @bobcatfish

/kind tep

Related PRs: 
- https://github.com/tektoncd/community/pull/494
- https://github.com/tektoncd/community/pull/170

### CODE REVIEW

**Comment by vdemeester** [comment]
@jerop what is the difference between this one and TEP-0078 ?

**Comment by jerop** [comment]
> @jerop what is the difference between this one and TEP-0078 ?

- TEP-0078 in https://github.com/tektoncd/community/pull/494 is about how we surface the information to the users in the Catalog, Hub, and CLI
- TEP-0079 in https://github.com/tektoncd/community/pull/495 is about how we actually establish the tiers with the automated testing and image scanning 

just broke it down so that we can move it along faster, but happy to merge them if that's easier

**Comment by vdemeester** [comment]
> > @jerop what is the difference between this one and TEP-0078 ?
> 
> * TEP-0078 in [TEP-0078: Tekton Catalog Support Tiers #494](https://github.com/tektoncd/community/pull/494) is about how we surface the information to the users in the Catalog, Hub, and CLI
> * TEP-0079 in [TEP-0079: Support Tiers in Tekton Catalog [Problem Statement] #495](https://github.com/tektoncd/community/pull/495) is about how we actually establish the tiers with the automated testing and image scanning
> 
> @vdemeester just broke it down so that we can move it along faster, but happy to merge them if that's easier

Ok make sense, then for the information 👼🏼

**Comment by pritidesai** [comment]
API WG - @jerop is updating the TEP - in progress

File: teps/0079-support-tiers-in-tekton-catalog.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+Today, all the resources in the Tekton Catalog are in the *community* support tier. To provide *verified* and *offical* 
+support tiers, we propose adding:
+
+1. Automated testing of resources in the Tekton Catalog using Tekton.
+1. Regular scanning of used images for common vulnerabilities and exposures
+1. Maintainance and ownership requirements for resources in the Tekton Catalog
+
+## Motivation
+
+### Goals
```
[the remark this answers is not in the document]
Comment (jerop) [reply teps/0079-support-tiers-in-tekton-catalog.md]: added support for tep-0091 to the scope of this tep - cc @squee1945

File: teps/0079-support-tiers-in-tekton-catalog.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+#### Maintenance and Ownership
+
+Every resource in the *Tekton Hub* needs to have owners to maintain them. Some resources will be maintained by Tekton
+community members, while other resources will be officially provided and maintained by Tekton maintainers.
+
+#### Automated Testing
+
+If we want users to rely on shared Tekton resources, we need to be able to make sure that they work. And if we want 
+contributors to share resources, we need to let them know how to provide tests that make sure that they work, and 
+we need to provide them with infrastructure that they can test against.
```
[the remark this answers is not in the document]
Comment (jerop) [reply teps/0079-support-tiers-in-tekton-catalog.md]: makes sense, added more details about dogfooding to the goals and also the maintainer user story

File: teps/0079-support-tiers-in-tekton-catalog.md [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+   1. If a resource is in the catalog, a user should feel confident to rely on them to work as advertised, so:
+      1. Linting and configuration tests should be applied to all resources
+      2. Testing should be applied regularly to all resources in the catalog
+      3. Contributors of resources should be able to easily run these tests themselves
+      4. It must be possible to apply this testing to resources that rely on external services (e.g. S3, slack)
+      5. Eventually tests should be run against varied Tekton installations, e.g. Tekton on different cloud providers
+   2. If a resource is offically provided by Tekton, the resources should be secure and owned by Tekton maintainers, so:
+      1. Images used in official resources must be regularly scanned for common vulnerabilities and exposures 
+      2. Tekton maintainers should own these resources so they can quickly respond to any issues regarding official 
+      resources
```
[the remark this answers is not in the document]
Comment (jerop) [reply teps/0079-support-tiers-in-tekton-catalog.md]: this section was starting to capture the requirements for official resources, so removed them from here and will expound on them in the proposal + include the demonstration of best practices

[the remark this answers is not in the document]
Comment (jerop) [reply teps/0079-support-tiers-in-tekton-catalog.md]: included the best practices in the maintainer user journey and goals instead

**Comment by jerop** [comment]
@bobcatfish @chmouel @vdemeester @vinamra28 Thank you for the initial reviews on Tekton Catalog Support Tiers - this TEP is ready for another look 😀 

@squee1945 highlighting that we added some goals in this TEP that are related to TEP-0091 (thanks to @bobcatfish!)

**Comment by vinamra28** [comment]
/approve
Looks fine from my side

**Pull request merged**